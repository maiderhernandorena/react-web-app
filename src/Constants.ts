export const AVATAR= 'https://www.pngkit.com/png/full/115-1150342_user-avatar-icon-iconos-de-mujeres-a-color.png'

export default {
    AVATAR
}