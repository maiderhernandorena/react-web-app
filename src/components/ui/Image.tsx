import React from 'react';

type ImageLayout = {
  width?: string
  height?: string
}

type ImageProps = {
  source: string,
  layout?: ImageLayout
}

const Image: React.FC<ImageProps> = ({ source, layout }) => {
  return (
        <img src={source} width={layout?.width} height={layout?.height}/>
  );
}

Image.defaultProps = {
  layout: {
    width: '100',
    height: '100'
  }
} as Partial<ImageProps>;

export default Image;
