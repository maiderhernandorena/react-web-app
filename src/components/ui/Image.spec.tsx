import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Image from './Image';
import { AVATAR } from '../../Constants'

describe('Image Component', () => {

  describe('Snapshots', () => {
    it('Should render with default size', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Image source={AVATAR} />
      );
      expect(wrapper).toMatchSnapshot();
    });
    
    it('Should render with specific size', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Image source={AVATAR} layout={{ width: '200', height: '200' }}/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render with only specific with', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Image source={AVATAR} layout={{ width: '200' }}/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render with only specific height', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Image source={AVATAR} layout={{ height: '200' }}/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});