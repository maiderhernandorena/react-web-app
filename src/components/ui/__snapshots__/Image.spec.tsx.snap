// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component Snapshots Should render with default size 1`] = `
<img
  height="100"
  src="https://www.pngkit.com/png/full/115-1150342_user-avatar-icon-iconos-de-mujeres-a-color.png"
  width="100"
/>
`;

exports[`Image Component Snapshots Should render with only specific height 1`] = `
<img
  height="200"
  src="https://www.pngkit.com/png/full/115-1150342_user-avatar-icon-iconos-de-mujeres-a-color.png"
/>
`;

exports[`Image Component Snapshots Should render with only specific with 1`] = `
<img
  src="https://www.pngkit.com/png/full/115-1150342_user-avatar-icon-iconos-de-mujeres-a-color.png"
  width="200"
/>
`;

exports[`Image Component Snapshots Should render with specific size 1`] = `
<img
  height="200"
  src="https://www.pngkit.com/png/full/115-1150342_user-avatar-icon-iconos-de-mujeres-a-color.png"
  width="200"
/>
`;
